### PR TITLE
Stop WooPay opt-in from blocking the place order button

### LINKED ISSUE #9136: WooPay "Save my info" when checked disables the "Place order" button
### Describe the bug
The checkout form by design keep the Place Order button in an enabled state so that shoppers can trigger validation errors. When the "save my info" checkbox is checked, the Place Order button is put into a disabled state. With pre-checking enabled by default, the impact of this becomes hugely amplified as form validation is blocked on page load.

### To Reproduce
1. On checkout, check the "save my info" checkbox and note that the Place Order button is disabled.

### Expected behavior
1. Checking "save my info" does not alter the core behavior of the form i.e. place order state is not altered and is active

### Additional context
Applies to shortcode and Blocks checkouts

### PR DESCRIPTION
**Pull request by leonardola**
Fixes #9136

#### Changes proposed in this Pull Request
This PR prevents the WooPay opt-in from blocking the place order button while the phone number was not typed

#### Testing instructions
- Checkout to this branch
- As a guest shopper add a product to the cart
- Go to the checkout page (test both blocks and shortcode)
- Select the `Securely save my information for 1-click checkout` checkbox
- Make sure no error shows up below the input field and the place order button is not blocked
- Fill in all required fields but the WooPay phone number
- Click the place order button
- Make sure the phone input error shows up and the place order button is not blocked
- Type the phone number
- Make sure the error message is gone
- Click the place order button
- Make sure the success page shows up

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9148` or branch name `fix/9136-woopay-opt-in-blocking-place-order-button` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/9136-woopay-opt-in-blocking-place-order-button) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 84fd6d65dd82d16ce898732322956587092ef047
- Build time: 2024-07-27 00:48:59 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/woopay/save-user/checkout-page-save-user.js [hunk 1]
```diff
@@ -85,7 +95,7 @@ const CheckoutPageSaveUser = ( { isBlocksCheckout } ) => {
 			extensionCartUpdate( {
 				namespace: 'woopay',
 				data: data,
-			} ).then( () => {
+			} )?.then( () => {
```
Comment (leonardola) [inline client/components/woopay/save-user/checkout-page-save-user.js:98]: This was the only solution I found to make the tests pass. Somehow this function was being called after the mocks where destroyed.

**Review by alefesouza — APPROVED** [review]
LGTM, thanks

**Review by hsingyuc — APPROVED** [review]
@leonardola Thank you for working on this issue! I can reproduce the issue when using theme Storefront but not Twenty-twenty-four and on the Blocks checkout page only.  

Test as expected and I'm going to merge this PR for Monday WooPayments release.

**Comment by hsingyuc** [comment]
I didn't refresh the page and just realized Alefe already reviewed and merged this PR. Thank you, @alefesouza.

**Comment by c-shultz** [comment]
Thanks all for pushing this through. It was great to get an extra look at it as well @hsingyuc

**Pull request merged**